src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/AdjournmentCalculateDateHelper.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.hmcts.reform.sscs.helper.mapping;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.beanutils.BeanUtils;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Optional;
+import javax.validation.Valid;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+@Slf4j
+public final class AdjournmentCalculateDateHelper {
+
+    static final int DAYS_TO_ADD_HEARING_WINDOW_TODAY_ADJOURNMENT = 14;
+
+    private AdjournmentCalculateDateHelper() {
+    }
+
+    static LocalDate getHearingWindowStart(@Valid SscsCaseData caseData) {
+        checkForMissingData("adjournCaseNextHearingDateType", caseData);
+        // TODO add to "Other" field time details from adjournCaseTime
+        String adjournCaseNextHearingDateType = caseData.getAdjournCaseNextHearingDateType();
+        Optional<AdjournmentDateType> adjournmentDateType = findDateTypeByValue(adjournCaseNextHearingDateType);
+        if (adjournmentDateType.isPresent()) {
+            return adjournmentDateType.get().resolveDate(caseData);
+        }
+        throw new IllegalArgumentException(String.format(
+            "Unexpected adjournCaseNextHearingDateType: %s for case id %s",
+            adjournCaseNextHearingDateType,
+            caseData.getCcdCaseId())
+        );
+    }
+
+    static LocalDate resolveDateForDateAfter(SscsCaseData caseData) {
+        String adjournCaseNextHearingDateOrPeriod = caseData.getAdjournCaseNextHearingDateOrPeriod();
+        checkForMissingData("adjournCaseNextHearingDateOrPeriod", caseData);
+        Optional<AdjournmentAfterDateOrPeriod> adjournmentAfterDateOrPeriod =
+            findDateOrPeriodByValue(adjournCaseNextHearingDateOrPeriod);
+        if (adjournmentAfterDateOrPeriod.isPresent()) {
+            return adjournmentAfterDateOrPeriod.get().resolveDate(caseData);
+        }
+        throw new IllegalArgumentException(String.format(
+            "Unexpected adjournCaseNextHearingDateOrPeriod: %s for case id %s",
+            adjournCaseNextHearingDateOrPeriod,
+            caseData.getCcdCaseId())
+        );
+    }
+
+    static Optional<AdjournmentDateType> findDateTypeByValue(String value) {
+        return Arrays.stream(AdjournmentDateType.values())
+            .filter(dateType -> dateType.value.equals(value))
+            .findFirst();
+    }
+
+    static Optional<AdjournmentAfterDateOrPeriod> findDateOrPeriodByValue(String value) {
+        return Arrays.stream(AdjournmentAfterDateOrPeriod.values())
+            .filter(dateOrPeriod -> dateOrPeriod.value.equals(value))
+            .findFirst();
+    }
+
+    @SneakyThrows
```

Comment:
I put this to avoid having to add ```throws RuntimeException``` to every single method up the hierarchy (and there are many). Is that in fact the correct thing to do here, or is there a better approach?

Reply:
have used try/catch to throw runtime